docs/website/02_examples/incompressibleRubberBlock.md

```diff
@@ -0,0 +1,120 @@
+The energy is then calculated as follows:
+```cpp
+energy += (0.5 * (2 * mu_ * symgradu.squaredNorm() - 1 / lambdaMat * Dune::power(pressure, 2)) + pressure * divU
+           - x.dot(fext))
+          * geo.integrationElement(gp.position()) * gp.weight();  // plane strain for 2D
+```
+Here: 
+- `symgradu` is the symmetric part of the gradient of displacements 
+- `lambdaMat` is the first Lame parameter
+- `pressure` and `x` are the nodal pressure and displacements, respectively
+- `divU` is the divergence of the displacement vector
+- `fext` is the external force vector
+- `gp.position()` and `gp.weight()` are the positions and weights from the quadrature rule
+- `geo.integrationElement()` returns the determinant of Jacobian required from the iso-parametric concept
+
+A Yasp 2D grid[@sander2020dune] is created of the size $L$ x $h$ with 20 elements in both directions as shown below: 
+```cpp
+using Grid        = Dune::YaspGrid<gridDim>;
+const double L    = 1;
+const double h    = 1;
+const size_t elex = 20;
+const size_t eley = 20;
+
+Dune::FieldVector<double, 2> bbox       = {L, h};
+std::array<int, 2> elementsPerDirection = {elex, eley};
+auto grid                               = std::make_shared<Grid>(bbox, elementsPerDirection);
+auto gridView                           = grid->leafGridView();
+```
+A linear Lagrangian basis is opted for the displacements and a constant basis for the pressure degrees of freedom using the
+`composite` basis feature from Dune as shown below:
+```cpp
+auto basis = Ikarus::makeConstSharedBasis(
+gridView, composite(power<2>(lagrange<1>(), FlatInterleaved()), lagrange<0>(), FlatLexicographic()));
+```
+The displacement degrees of freedom at position $y=0$ are fixed using the following snippet:
+```cpp
+Ikarus::DirichletValues dirichletValues(basis);
+dirichletValues.fixDOFs([](auto &basis_, auto &dirichletFlags) {
+  Dune::Functions::forEachBoundaryDOF(subspaceBasis(basis_, _0),
+                   [&](auto &&localIndex, auto &&localView, auto &&intersection) {
+                       if (std::abs(intersection.geometry().center()[1]) < 1e-8)
+                         dirichletFlags[localView.index(localIndex)] = true;
+  });
+}); 
+```
+Here, all the element edges lying on the boundary of the physical domain are looped over and checked to see if the first index 
+of the center of the edge (`intersection.geometry().center()[1]`) is closer to zero. If this is the case, the corresponding displacement 
+degrees of freedom (obtained via `subspaceBasis(basis_, _0)`) are set to `true` and used by the assembler later.
+
+A `sparse` and a `dense` assembler are used to arrive at the stiffness matrix and the external load vector using the 
+finite element requirements as described [here](../01_framework/feRequirements.md#fe-requirements).  
+```cpp
+auto sparseFlatAssembler = SparseFlatAssembler(fes, dirichletValues);
+auto denseFlatAssembler  = DenseFlatAssembler(fes, dirichletValues);
```

Comment:
ok this is also outdated code. It should also work with only the `SparseFlatAssembler`. But then we have to also edit the example